newIDE/app/src/EventsSheet/EventsTree/DropContainer.js

```diff
@@ -0,0 +1,323 @@
+              dropIndicator: indicatorStyles['bottom-right'],
+              dropArea: dropAreaStyles['bottom-right'],
+            }}
+            onDrop={() => onDrop(moveNodeAsSubEvent, node)}
+            {...commonProps}
+          />
+          <DropTargetContainer
+            style={{
+              dropIndicator: indicatorStyles['bottom'],
+              dropArea: dropAreaStyles['bottom-left'],
+            }}
+            onDrop={() => onDrop(moveNodeBelow, node)}
+            {...commonProps}
+          />
+          {/* Allow dragging left/right to move below or as subevent */}
+          {(isDraggedNodesOnlyChild || isDraggedNodeSameDepthAndBelow) && (
+            <>
+              <DropTargetContainer
+                style={{
+                  dropIndicator: indicatorStyles['bottom'],
+                  dropArea: dropAreaStyles['below-left'],
+                }}
+                onDrop={() => onDrop(moveNodeBelow, node)}
+                {...commonProps}
+              />
+              <DropTargetContainer
+                style={{
+                  dropIndicator: indicatorStyles['bottom-right'],
+                  dropArea: dropAreaStyles['below-right'],
+                }}
+                onDrop={() => onDrop(moveNodeAsSubEvent, node)}
+                {...commonProps}
+              />
+            </>
+          )}
+        </>
+      )}
+      {!canHaveSubEvents && canDropBelow && (
+        <DropTargetContainer
+          style={{
+            dropIndicator: indicatorStyles['bottom'],
+            dropArea: dropAreaStyles['bottom'],
+          }}
+          onDrop={() => onDrop(moveNodeBelow, node)}
+          {...commonProps}
+        />
+      )}
+      {canHaveSubEvents && !canDropBelow && (
+        <DropTargetContainer
+          style={{
+            dropIndicator: indicatorStyles['bottom-right'],
+            dropArea: dropAreaStyles['bottom'],
+          }}
+          onDrop={() => onDrop(moveNodeAsSubEvent, node)}
+          {...commonProps}
+        />
+      )}
+    </div>
+  );
+}
+
+export function AutoScroll({
+  direction,
+  DnDComponent,
+  activateTargets,
+  onHover,
+}: {|
+  direction: 'top' | 'bottom',
+  DnDComponent: DropTargetComponent<SortableTreeNode>,
+  activateTargets: boolean,
+  onHover: () => void,
+|}) {
+  const delayActivationTimer = React.useRef<?TimeoutID>(null);
+  const [show, setShow] = React.useState(false);
+
+  // This drop target overlaps with sibling drag source and cancels drag immediately.
+  // See: https://github.com/react-dnd/react-dnd/issues/766#issuecomment-388943403
+  // Delaying the render of the drop target seems to solve the issue.
+  React.useEffect(
+    () => {
+      if (activateTargets) {
+        delayActivationTimer.current = setTimeout(() => {
+          setShow(true);
+        }, 100);
+      } else {
+        setShow(false);
+        clearTimeout(delayActivationTimer.current);
+        delayActivationTimer.current = null;
+      }
+    },
```

Comment:
Should you return a cleanup function from this effect? Is there a risk that unmounting the component would not clear the timeout? (it's 100ms so the risk is minimal, just give it a try to see if using the cleanup function of the effect works)